lib/dao.js

```diff
@@ -1183,8 +1191,9 @@ DataAccessObject._coerce = function (where) {
       }
     } else {
       if (val != null) {
-        if (!((operator === 'like' || operator === 'nlike') &&
-          val instanceof RegExp)) {
+        if (val instanceof RegExp) {
+          // do not coerce regex literals/objects
+        } else if (!((operator === 'like' || operator === 'nlike') && val instanceof RegExp)) {
```

Comment:
@raymondfeng This is the culprit. How do you recommend I clean this up without breaking old code? My implementation with the empty if check for RegExp works, but I'm not sure if this is the best solution. The other possibility is to fix `!((operator === 'like' || operator === 'nlike') && val instanceof RegExp))`.